erdjs/src/scArgumentsParser.ts

```diff
@@ -0,0 +1,70 @@
+import {ErrInvalidScCallDataField} from "./errors";
+
+/**
+ * Class with static methods useful for fetching and checking arguments from a transaction's data field that should trigger
+ * a smart contract call
+ */
+export class ScArgumentsParser {
+
+    private static validHexChars = ["a", "b", "c", "d", "e", "f", "0", "1", "2", "3", "4", "5", "6", "7", "8", "9"];
+
+    /**
+     * Returns an array containing all the arguments from a data field representing a smart contract call
+     * @param dataField dataField represents the data filed to extract arguments from
+     */
+    public static getArgumentsFromDataField(dataField: string): Array<string> {
+        if (!this.isValidDataField(dataField)) {
+            throw new ErrInvalidScCallDataField(dataField);
+        }
+
+        let args = new Array<string>();
+        let items = dataField.split("@");
+        if (items.length == 0) {
+            args.push(dataField);
+            return args; // no argument, return the array containing only the data field
+        }
```

Comment:
If there are no arguments to the smart contract call, I think this function should return an empty array. Why return the entire dataField as a first argument?

Reply:
refactored so the function returns both the function name and the arguments